src/Config.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\HttpExtractor;
+
+use Keboola\HttpExtractor\Config\ConfigDefinition;
+use Symfony\Component\Config\Definition\Processor;
+use Symfony\Component\Serializer\Encoder\JsonDecode;
+use Symfony\Component\Serializer\Encoder\JsonEncoder;
+
+class Config
+{
+    /** @var array */
+    private $config;
+
+    public function __construct(
+        array $config
+    ) {
+        $definition = new ConfigDefinition();
+        $processor = new Processor();
+        $processedConfig = $processor->processConfiguration($definition, [$config]);
+        $this->config = $processedConfig;
+    }
+
+    public static function fromFile(string $configFilePath): self
+    {
+        $contents = file_get_contents($configFilePath);
+        $decoder = new JsonDecode(true);
+        $config = $decoder->decode($contents, JsonEncoder::FORMAT);
+
+        return new self($config['parameters']);
+    }
+
+    public function getData(): array
+    {
+        return $this->config;
+    }
+
+    public function getDownloadUrlBase(): string
```

Comment:
Jsou tady ty gettery vůbec potřeba? Kdyby to mělo jen ctor, `fromFile` a `getData`, tak by to mohlo bejt univerzálně použitelný pro všechny PHP komponenty.

Reply:
Ty gettery jsou proto, aby sis nemusel pamatovat klíče pole a nemohl v nich udělat typo. 

Jako ideálně by to byla knihovna. 

Uživatel by si udělal `Config extends KeboolaConfig`. `KeboolaConfig` by byl abstraktní s metodou `getConfigDefinition(): ConfigurationInterface`, ze které by se vytahla ta secka toho konfiguráku a použila se v konstruktoru pro ověření. K tomu by byl abstraktní `KeboolaConfigDefinition`, který by měl v sobě základní strukturu toho keboolího configu (celého). A obsahoval by protected `getParametersDefinition()`, `getStorageInput` a `getStorageOutput` - tak aby šly přetížit, pokud je někdo potřebuje nějak konkrétně specifikovat (případně by mohly být abstraktní + privátní `getDefaultStorageInput()` pokud člověk nechce nic měnit.  A nakonec `ConfigLoader`, který by měl `loadConfig($path, Config::class)` z cesty by natahnul JSON, ten zkontroloval, dekódoval a udělal z něj config. Uživatel by pak jen doimplementoval convenience gettery, aby mu to napovídalo datové typy a metody a nemusel si pamatovat klíče pole. 

Přišlo by mi to správně, ale vzhledem k tomu, že jste k ničemu takovému nedokonvergovali, tak jsem si říkal, že pro to bude nějaký důvod. A že to chcete držet víc obecně bez nějakého frameworku okolo. Což jsme se konec konců bavili v těhle intencích i okolo Syrupu.

Reply:
my jsme k tomu uz celkem dokonvergovali https://github.com/keboola/python-docker-application a https://github.com/keboola/r-docker-application ale ne pro PHP :)

Reply:
To by bylo super mít něco takovýho! S tím pamatováním a convenience getterama souhlas.